tests/vm_test.go

```diff
@@ -298,23 +284,14 @@ var _ = Describe("[rfe_id:1177][crit:medium][vendor:cnv-qe@redhat.com][level:com
 		})
 
 		It("[test_id:3162]should ignore kubernetes and kubevirt annotations to VMI", func() {
-			annotations := map[string]string{
+			vm := libvmi.NewVirtualMachine(libvmifact.NewGuestless(), libvmi.WithRunning())
+			vm.Annotations = map[string]string{
 				"kubevirt.io/test":   "test",
 				"kubernetes.io/test": "test",
 			}
-
-			vm := createVM(virtClient, libvmifact.NewCirros())
-
-			err = tests.RetryWithMetadataIfModified(vm.ObjectMeta, func(meta k8smetav1.ObjectMeta) error {
-				vm, err = virtClient.VirtualMachine(meta.Namespace).Get(context.Background(), meta.Name, k8smetav1.GetOptions{})
-				Expect(err).ToNot(HaveOccurred())
-				vm.Annotations = annotations
-				vm, err = virtClient.VirtualMachine(meta.Namespace).Update(context.Background(), vm, metav1.UpdateOptions{})
-				return err
-			})
+			vm, err := virtClient.VirtualMachine(testsuite.GetTestNamespace(vm)).Create(context.Background(), vm, k8smetav1.CreateOptions{})
```

Comment:
Can you drop all changes of this test then? It is confusing to see it here changed and later the test removed completely.

Reply:
I disagree. This change caused the test to fail, and the later commit removed it (consider it as a solution).

Reply:
Actually I thought about it before pushing the new commit. I asked myself, how would I explain removing "working" functional test, before my changes caused the failure of it?

Reply:
> how would I explain removing "working" functional test

In your commit message you should explain the history, as revealed by @akalenyu: that the tested code was removed, and that the test used to pass only by accident.

Reply:
I do not understand why as a reviewer I need to be explain all this history about refactoring causing a test to be more clear and show it is not needed. Couldn’t you reason the same before cleaning the test?

You are basically asking all reviewers to invest energy in understanding a change which later is removed. If one walks the commits it is a unnecessary burden and if one looks at the unified PR, it will not be noticed.

I do not think it is a good thing, but it is not worth getting stuck on it.